bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/handler/IPBridgeHandler.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.alarmdecoder.internal.handler;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.alarmdecoder.internal.config.IPBridgeConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler responsible for communicating via TCP with the Nu Tech Alarm Decoder device.
+ * Based on and including code from the original OH1 alarmdecoder binding.
+ *
+ * @author Bernd Pfrommer - Initial contribution (OH1 version)
+ * @author Bob Adair - Re-factored into OH2 binding
+ */
+@NonNullByDefault
+public class IPBridgeHandler extends ADBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IPBridgeHandler.class);
+
+    private IPBridgeConfig config = new IPBridgeConfig();
+
+    private @Nullable Socket socket = null;
+
+    public IPBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IP bridge handler");
+        config = getConfigAs(IPBridgeConfig.class);
+        discovery = config.discovery;
+
+        if (config.hostname == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "hostname not configured");
+            return;
+        }
+        if (config.tcpPort <= 0 || config.tcpPort > 65535) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "invalid port number configured");
+            return;
+        }
+
+        // set the thing status to UNKNOWN temporarily and let the background connect task decide the real status.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.submit(this::connect); // start the async connect task
+    }
+
+    @Override
+    protected synchronized void connect() {
+        disconnect(); // make sure we are disconnected
+        writeException = false;
+        try {
+            socket = new Socket(config.hostname, config.tcpPort);
```

Comment:
For instance, the warnings on the following two lines can be fixed like this:
```suggestion
            Socket socket = new Socket(config.hostname, config.tcpPort);
            this.socket = socket;
```